src/ZkFold/Base/Algebra/Basic/Class.hs

```diff
@@ -64,8 +61,12 @@ class (AdditiveMonoid a, MultiplicativeMonoid a, FromConstant Natural a) => Semi
 
 class (Semiring a, AdditiveGroup a, FromConstant Integer a) => Ring a
 
+data Characteristic = Infinite | Finite Natural
+
 -- NOTE: by convention, division by zero returns zero.
-type Field a = (Ring a, MultiplicativeGroup a)
+class (Ring a, MultiplicativeGroup a) => Field a where 
+    rootOfUnity :: Natural -> Maybe a
+    rootOfUnity = const Nothing
```

Comment:
Every field has a `rootOfUnity 0 = Just one` by an axiom `zero /= one` (though we might not need this)